tableschema_sql/mapper.py

```diff
@@ -192,3 +212,53 @@ def restore_descriptor(self, table_name, columns, constraints, autoincrement_col
             descriptor['foreignKeys'] = fks
 
         return descriptor
+
+    def restore_row(self, row, schema):
+        """Restore row from SQL
+        """
+        return schema.cast_row(row)
+
+    def restore_type(self, type):
+        """Restore type from SQL
+        """
+
+        # All dialects
+        mapping = {
+            ARRAY: 'array',
+            sa.Boolean: 'boolean',
+            sa.Date: 'date',
+            sa.DateTime: 'datetime',
+            sa.Float: 'number',
+            sa.Integer: 'integer',
+            JSONB: 'object',
+            JSON: 'object',
+            sa.Text: 'string',
+            sa.Time: 'time',
+            sa.VARCHAR: 'string',
+            UUID: 'string',
+        }
+
+        # Get field type
+        field_type = None
+        for key, value in mapping.items():
+            if isinstance(type, key):
+                field_type = value
+
+        # Not supported
+        if field_type is None:
+            message = 'Type "%s" is not supported'
+            raise TypeError(message % type)
+
+        return field_type
+
+
+# Internal
+
+def uncast_value(value, field):
+    # Eventially should be moved to:
+    # https://github.com/frictionlessdata/tableschema-py/issues/161
+    if isinstance(value, (list, dict)):
+        value = json.dumps(value)
+    else:
+        value = str(value)
```

Comment:
This doesn't necessarily generate the 'default' formatting for types.
Specifically for time types, python will not add the final `.fff` (microseconds) if it's 0, and will add it in case it's not.

Reply:
Yes. For now it's a minimal implementation because for this driver we never have to stringify something other than `list/dict` or types that work with `str(value)`. I'm going to write it properly in https://github.com/frictionlessdata/tableschema-py/issues/161 (supporting format etc) because all `tableschema` plugins need this function.